src/store/inflight_activation.rs

```diff
@@ -166,66 +192,111 @@ struct TableRow {
+            activation: value.activation,
             status: value.status,
             partition: value.partition,
             offset: value.offset,
             added_at: value.added_at,
+            received_at: value.received_at,
             processing_attempts: value.processing_attempts,
+            processing_deadline_duration: value.processing_deadline_duration,
             expires_at: value.expires_at,
             delay_until: value.delay_until,
             processing_deadline: value.processing_deadline,
             at_most_once: value.at_most_once,
             namespace: value.namespace,
+            taskname: value.taskname,
+            on_attempts_exceeded: value.on_attempts_exceeded,
         }
     }
 }
 
+async fn create_sqlite_pool(url: &str) -> Result<(Pool<Sqlite>, Pool<Sqlite>), Error> {
+    if !Sqlite::database_exists(url).await? {
+        Sqlite::create_database(url).await?
+    }
+
+    let read_pool = PoolOptions::<Sqlite>::new()
+        .max_connections(64)
+        .connect_with(
+            SqliteConnectOptions::from_str(url)?
+                .journal_mode(SqliteJournalMode::Wal)
+                .synchronous(SqliteSynchronous::Normal)
+                .read_only(true)
+                .disable_statement_logging(),
+        )
+        .await?;
+
+    let write_pool = PoolOptions::<Sqlite>::new()
+        .max_connections(1)
+        .connect_with(
+            SqliteConnectOptions::from_str(url)?
+                .journal_mode(SqliteJournalMode::Wal)
+                .synchronous(SqliteSynchronous::Normal)
+                .auto_vacuum(SqliteAutoVacuum::Incremental)
+                .disable_statement_logging(),
+        )
+        .await?;
+
+    Ok((read_pool, write_pool))
+}
+
+pub struct InflightActivationStoreConfig {
+    pub max_processing_attempts: usize,
+}
+
+impl InflightActivationStoreConfig {
+    pub fn from_config(config: &Config) -> Self {
+        Self {
+            max_processing_attempts: config.max_processing_attempts,
+        }
+    }
+}
+
+pub struct InflightActivationStore {
+    read_pool: SqlitePool,
+    write_pool: SqlitePool,
+    config: InflightActivationStoreConfig,
+}
+
 impl InflightActivationStore {
     pub async fn new(url: &str, config: InflightActivationStoreConfig) -> Result<Self, Error> {
-        if !Sqlite::database_exists(url).await? {
-            Sqlite::create_database(url).await?
-        }
+        let (read_pool, write_pool) = create_sqlite_pool(url).await?;
 
-        let read_pool = PoolOptions::<Sqlite>::new()
-            .max_connections(64)
-            .connect_with(
-                SqliteConnectOptions::from_str(url)?
-                    .journal_mode(SqliteJournalMode::Wal)
-                    .synchronous(SqliteSynchronous::Normal)
-                    .read_only(true)
-                    .disable_statement_logging(),
-            )
-            .await?;
+        sqlx::migrate!("./migrations").run(&write_pool).await?;
```

Comment:
We also run migrations on line 312 do we need to do it twice?

Reply:
Oops bad merge.